### ISSUE TITLE
Prevent deleting a branch when there are PR's based on that branch.

### ISSUE THREAD

**Issue by CarloPalinckx** [chdsbd/kodiak#177]
It would be nice if there was some way of preventing kodiak to remove the merged branch if any PR depends on it (like the github UI does). Currently, based PR's get closed and the remote branch of that PR get's closed.

**Comment by chdsbd** [comment]
Is it possible to do this through the GitHub UI right now?

The GitHub API is a pretty close match to the operations available through the UI, so my thought is that if it's not possible to do this through the GitHub UI, it's probably not possible to do through the API.

**Comment by CarloPalinckx** [comment]
Github UI does this automatically, there is no opt-out for this if I'm not mistaken.

**Comment by chdsbd** [comment]
Could you provide an example that shows how Kodiak should behave and how it actual behaves? I'm not sure I understand your branch and pull request setup.

**Comment by chdsbd** [comment]
Looking at this again, I think the issue is that with `merge.delete_branch_on_merge = true`, Kodiak will delete the branch of the PR on merge. However, there could be other PRs based on the same branch, which Kodiak would incidentally destroy by deleting the shared branch.

I think we could fix this by checking if there are other PRs opened with the same head ref with a query like the following.

```graphql
query { 
  repository(name:"kodiak", owner:"chdsbd") {
    databaseId
    pullRequest(number:227) {
      id
      baseRefName
      headRef {
        associatedPullRequests(first: 100) {
          nodes {
            id
          }
        }
      }
    }
  }
} 
```

Although considering that GitHub has a setting to delete branches on merge, we might be able to get away with deprecating the redundant `merge.delete_branch_on_merge` Kodiak config setting.

![image](https://user-images.githubusercontent.com/1929960/72005150-e8883300-321a-11ea-8541-2b697d300a61.png)

**Comment by chdsbd** [comment]
I was not able to fix this in #232 and had to revert that change with #237.

It seems that there isn't a simple way through the GitHub api to see PRs that depend on a ref, so I'm not sure how to fix this at the moment.
